docs/examples/check-email-for-intent.md

```diff
@@ -0,0 +1,243 @@
+      port: 587
+  transition: sendcloudevent
+```
+
+### Trigger Event
+This action state uses the request container to send a cloud event back to Direktiv.
+
+```yaml
+- id: sendcloudevent
+  type: action
+  action: 
+    function: request
+    input:
+      method: "POST"
+      url: "http://192.168.1.30/api/namespaces/direktiv/event"
+      body: 
+        id: "read-email-message"
+        specversion: "1.0"
+        type: "smtp"
+```
+
+### Workflow Send Email 
+
+```yaml
+id: send-email-trigger-event
+functions:
+- id: request
+  image: vorteil/request:v5
+- id: smtp
+  image: vorteil/smtp:v3
+description: This workflow sends an email and triggers an event.
+states:
+- id: sendemail
+  type: action
+  action:
+    secrets: ["EMAIL_NAME", "EMAIL_PASSWORD"]
+    function: smtp
+    input:
+      to: "{{.secrets.EMAIL_NAME}}"
+      subject: "Your Review"
+      message: "You are very bad at doing work."
+      from: "{{.secrets.EMAIL_NAME}}"
+      password: "{{.secrets.EMAIL_PASSWORD}}"
+      server: smtp.gmail.com
+      port: 587
+  transition: sendcloudevent
+- id: sendcloudevent
+  type: action
+  action: 
+    function: request
+    input:
+      method: "POST"
+      url: "http://192.168.1.30/api/namespaces/direktiv/event"
+      body: 
+        id: "read-email-message"
+        specversion: "1.0"
+        type: "smtp"
+```
+
+
+## Read the Email and Check its Intent
+
+```yaml
+id: listen-for-email
+description: This workflow reads an email when a cloud event is received.
+functions:
+- id: imap
+  image: vorteil/imap:v1
+- id: smtp
+  image: vorteil/smtp:v3
+- id: sentiment
+  image: vorteil/google-sentiment-check:v2
+start:
+  type: event
+  state: read-email
+  event:
+    type: smtp
+states:
+  # continued in next code block
+```
+
+**NOTE: when the 'direktiv' namespace gets a cloud event of 'smtp' type this workflow will be triggered.**
+
+
+### Read Email
+This takes the first message from your "INBOX" email and reads the body and outputs it.
+
+```yaml
+- id: read-mail
+  type: action
+  action:
+    secrets: ["EMAIL_NAME", "EMAIL_PASSWORD"]
+    function: imap
+    input:
+      email: "{{.secrets.EMAIL_NAME}}"
+      password: "{{.secrets.EMAIL_PASSWORD}}"
+      imap-address: "imap.gmail.com:993"
+  transition: sentiment-check
+```
+
+### Check Intent
+
+```yaml
+- id: sentiment-check
+  type: action
+  action:
+    function: sentiment
+    secrets: ["SERVICE_ACCOUNT_KEY"]
+    input:
+      message: "{{.return.msg}}"
+      serviceAccountKey: "{{.secrets.SERVICE_ACCOUNT_KEY}}"
+  transition: check-feeling
+```
+
+### Check Feeling
+If the feeling of the text written in the email is negative send an automatic response back to the sender.
+
+```yaml
+- id: check-feeling
+  type: switch
+  conditions:
+  - condition: '.return.feeling == "Negative"'
+    transition: send-response
+```
+
+### Reply to Email
+This uses the smtp server to send a response to the person that sent you the email in the first place.
```

Comment:
This state uses the `vorteil/smtp:v2` isolate to send an email to the sender of the negative email.